test/tests/functional/pbs_hook_debug_input.py

```diff
@@ -84,6 +92,27 @@ def match_queue_name_in_input_file(self, input_file_pattern, qname):
             self.assertTrue(search_str in f.read().decode())
         self.remove_files_match(input_file_pattern)
 
+    def match_in_debug_file(self, input_file_pattern, search_list, mom=False):
+        """
+        Assert that all the strings in 'search_list' appears in the hook
+        debug file that matches input_file_pattern
+        """
+        input_file = None
+        if mom:
+            hooks_tmp_dir = self.mom_hooks_tmp_dir
+        else:
+            hooks_tmp_dir = self.server_hooks_tmp_dir
+        for item in self.du.listdir(path=hooks_tmp_dir, sudo=True):
+            if fnmatch.fnmatch(item, input_file_pattern):
+                input_file = item
+                break
+        self.assertTrue(input_file is not None)
+        with PBSLogUtils().open_log(input_file, sudo=True) as f:
+            content = f.read().decode()
+            for entry in search_list:
+                self.assertTrue(entry in content)
```

Comment:
How large, typically,  are the hook debug files, and the search list? This search process might be very slow.

Reply:
Typical with generic install of PBS is about 2700 bytes/59 lines:
```
ls -l *.data
-rw-r--r-- 1 root root 2740 Sep 15 12:39 hook_execjob_begin_debug_19633.data
``

Reply:
Can the data be decoded to a dictionary to make the search faster?

Reply:
@minghui-liu : the file is just a dump of pbs.server() data that is driven by the hook script content.
The following line in hook script:
s = pbs.server()

puts in the hook data file:
```
pbs.server().server_state=Active
pbs.server().server_host=corretja.pbspro.com
...:
```
and the hook content of:
q = pbs.server("workq")
generates:
```
pbs.server().queue(workq).queue_type=Execution
pbs.server().queue(workq).total_jobs=0
pbs.server().queue(workq).state_count=Transit:0 Queued:0 Held:0 Waiting:0 Running:0 Exiting:0 Begun:0 
pbs.server().queue(workq).enabled=True
pbs.server().queue(workq).started=True
```
and the
for vn in s.vnodes():
generates:
```
pbs.server().vnode(corretja).Mom=corretja.pbspro.com
pbs.server().vnode(corretja).Port=15002
pbs.server().vnode(corretja).pbs_version=2021.1.0.20200826072608
pbs.server().vnode(corretja).ntype=0
```
For this single-node test, no more than 3000 bytes of data would be produced. Each line is just some```<attribute>=<value>``` entries, where ```<attribute>``` could be "pbs.server()', "pbs.server().queue(<queue_name>)", or 'pbs.server().vnode(<vnode-name>)'. I would still need to loop through each entry if I want to convert the content into a dictionary.